src/HonzaBotner.Discord.Services/Commands/EmoteCommands.cs

```diff
@@ -1,70 +1,54 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
-using DSharpPlus.CommandsNext;
-using DSharpPlus.CommandsNext.Attributes;
+using DSharpPlus;
 using DSharpPlus.Entities;
 using DSharpPlus.Interactivity;
+using DSharpPlus.Interactivity.Enums;
 using DSharpPlus.Interactivity.Extensions;
-using HonzaBotner.Discord.Services.Extensions;
-using HonzaBotner.Discord.Services.Attributes;
+using DSharpPlus.SlashCommands;
 using HonzaBotner.Services.Contract;
 using HonzaBotner.Services.Contract.Dto;
-using Microsoft.Extensions.Logging;
 
 namespace HonzaBotner.Discord.Services.Commands;
 
-[Group("emotes")]
-[Aliases("emote", "emojis", "emoji")]
-[Description(
-    "Commands to display stats about emote usage. You can also use additional switches `animated` and `nonanimated`.")]
-[ModuleLifespan(ModuleLifespan.Transient)]
-[RequireMod]
-[RequireGuild]
-public class EmoteCommands : BaseCommandModule
+[SlashModuleLifespan(SlashModuleLifespan.Scoped)]
+public class EmoteCommands : ApplicationCommandModule
 {
-    private readonly IEmojiCounterService _emojiCounterService;
-    private readonly ILogger<EmoteCommands> _logger;
 
-    public EmoteCommands(IEmojiCounterService emojiCounterService, ILogger<EmoteCommands> logger)
-    {
-        _emojiCounterService = emojiCounterService;
-        _logger = logger;
-    }
+    private readonly IEmojiCounterService _emojiCounterService;
 
-    [GroupCommand]
-    [Command("perday")]
-    [Aliases("daily")]
-    [Description("Displays per day usage of emotes.")]
-    public Task PerDayCommand(CommandContext ctx, params string[] parameters)
+    public enum DisplayTypes
     {
-        return Display(ctx, false, parameters, emoji => emoji.UsagePerDay);
+        [ChoiceName("all")]
+        All,
+        [ChoiceName("animated")]
+        Animated,
+        [ChoiceName("still")]
+        Still
     }
 
-    [Command("total")]
-    [Aliases("all")]
-    [Description("Displays total usage of emotes.")]
-    public Task TotalCommand(CommandContext ctx, params string[] parameters)
+    public EmoteCommands(IEmojiCounterService emojiCounterService)
     {
-        return Display(ctx, true, parameters, emoji => emoji.Used);
+        _emojiCounterService = emojiCounterService;
     }
 
-    private async Task Display<TKey>(CommandContext ctx, bool total, string[] parameters,
-        Func<CountedEmoji, TKey> comparer)
+    [SlashCommand("emotes", "Display statistics about emotes on server")]
+    [SlashCommandPermissions(Permissions.ManageEmojis)]
+    public async Task EmoteStatsCommandAsync(
+        InteractionContext ctx,
+        [Option("display", "Display as total instead of perDay?")] bool total = true,
+        [Option("Type", "What type of emojis to show? Defaults all")] DisplayTypes type = DisplayTypes.All)
```

Comment:
Should the first parameter be unified, so it either starts with uppercase or lowercase?

Reply:
Will fix

Reply:
<@427441425861443595>